--- createrepo_c-0.10.0/doc/python/CMakeLists.txt.orig	2018-05-11 20:05:51.125805641 +0200
+++ createrepo_c-0.10.0/doc/python/CMakeLists.txt	2018-05-11 20:08:41.395803696 +0200
@@ -1,5 +1,5 @@
 ADD_CUSTOM_TARGET (doc-python
-    PYTHONPATH=${CMAKE_BINARY_DIR}/src/python sphinx-build -E -b html
+    PYTHONPATH=${CMAKE_BINARY_DIR}/src/python sphinx-build-2 -E -b html
                   ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/html
                   COMMENT "Building Python API documentation with Sphinx")
 
--- createrepo_c-0.10.0/src/python/CMakeLists.txt.orig	2018-05-11 20:05:51.125805641 +0200
+++ createrepo_c-0.10.0/src/python/CMakeLists.txt	2018-05-11 20:09:44.352469646 +0200
@@ -5,8 +5,8 @@ if (${PYTHON_DESIRED} STREQUAL "2")
     unset(PYTHON_LIBRARY CACHE)
     unset(PYTHON_INCLUDE_DIR CACHE)
     unset(PYTHON_EXECUTABLE CACHE)
-    FIND_PACKAGE(PythonLibs 2)
     FIND_PACKAGE(PythonInterp 2 REQUIRED)
+    FIND_PACKAGE(PythonLibs 2)
 else()
     SET(Python_ADDITIONAL_VERSIONS 3.0)
     unset(PYTHON_LIBRARY)
@@ -15,11 +15,11 @@ else()
     unset(PYTHON_LIBRARY CACHE)
     unset(PYTHON_INCLUDE_DIR CACHE)
     unset(PYTHON_EXECUTABLE CACHE)
-    FIND_PACKAGE(PythonLibs 3.0)
     FIND_PACKAGE(PythonInterp 3.0 REQUIRED)
+    FIND_PACKAGE(PythonLibs 3.0)
 endif()
 
-EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from distutils import sysconfig; stdout.write(sysconfig.get_python_lib(True))" OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
+EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from distutils import sysconfig; stdout.write(sysconfig.get_python_lib(True, prefix='/usr'))" OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
 INCLUDE_DIRECTORIES (${PYTHON_INCLUDE_PATH})
 
 MESSAGE(STATUS "Python install dir is ${PYTHON_INSTALL_DIR}")
