--- createrepo_c-0.21.1/src/cmd_parser.c.orig	2023-04-04 12:30:43.000000000 +0200
+++ createrepo_c-0.21.1/src/cmd_parser.c	2023-06-24 20:41:36.793262765 +0200
@@ -73,9 +73,9 @@ struct CmdOptions _cmd_options = {
 
 
 gboolean
-duplicated_nevra_option_parser(const gchar *,
+duplicated_nevra_option_parser(const gchar *optname_,
                                const gchar *value,
-                               gpointer,
+                               gpointer data_,
                                GError **error)
 {
     if (!g_strcmp0(value, "keep"))
